src/github-api-helper.ts

```diff
@@ -0,0 +1,88 @@
+import * as assert from 'assert'
+import * as core from '@actions/core'
+import * as fs from 'fs'
+import * as github from '@actions/github'
+import * as io from '@actions/io'
+import * as path from 'path'
+import * as retryHelper from './retry-helper'
+import * as toolCache from '@actions/tool-cache'
+import {default as uuid} from 'uuid/v4'
+import {ReposGetArchiveLinkParams} from '@octokit/rest'
+
+const IS_WINDOWS = process.platform === 'win32'
+
```

Comment:
pretty simple function:
- Download the archive
- Write to disk
- Extract
- Move the files to the correct place (the archive contains an extra top level folder that we need to flatten)